llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -1129,9 +1125,26 @@ void AMDGPUPassConfig::addCodeGenPrepare() {
     // but has been put before switch lowering and CFG flattening so that those
     // passes can run on the more optimized control flow this pass creates in
     // many cases.
+    //
+    // FIXME: This should ideally be put after the LoadStoreVectorizer.
```

Comment:
Looks ok as a workaround of https://github.com/llvm/llvm-project/issues/64863, and agreed that the previous placement of the new pass was preferred.